### PR TITLE
fix: add revenue props in the event payload when there has revenue or…

### PR DESCRIPTION
**Pull request by yuhao900914**
… quantity and price info

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Java/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

File: src/main/java/com/amplitude/Event.java [hunk 1]
```diff
@@ -240,11 +240,18 @@ public JSONObject toJsonObject() {
...
-      event.put("productId", productId);
-      event.put("revenueType", revenueType);
+
+      boolean shouldLogRevenueProps = (revenue != null || (price != null && quantity > 0));
+      if (shouldLogRevenueProps) {
+        event.put("price", price);
+        event.put("quantity", quantity);
+        event.put(
+            "revenue",
+            (revenue != null && price != null && quantity > 0) ? quantity * price : revenue);
```
Comment (qingzhuozhen) [inline src/main/java/com/amplitude/Event.java:250]: I am a bit confused here, so if there exists revenue and price and quantity then we do our calculation?

File: src/main/java/com/amplitude/Event.java [hunk 2]
```diff
@@ -240,11 +240,18 @@ public JSONObject toJsonObject() {
       event.put(
           "user_properties",
           (userProperties == null) ? new JSONObject() : truncate(userProperties));
-      event.put("price", price);
-      event.put("quantity", quantity);
-      event.put("revenue", revenue);
-      event.put("productId", productId);
-      event.put("revenueType", revenueType);
+
+      boolean shouldLogRevenueProps = (revenue != null || (price != null && quantity > 0));
```
Comment (qingzhuozhen) [inline src/main/java/com/amplitude/Event.java:244]: Could we add a simple test case for this?

File: src/main/java/com/amplitude/Event.java (hunk 1 above)
Comment (yuhao900914) [reply src/main/java/com/amplitude/Event.java:250]: it’s how api v2 says.

https://developers.amplitude.com/docs/http-api-v2

revenue = price * quantity. If you send all 3 fields of price, quantity, and revenue, then (price * quantity) will be used as the revenue value. You can use negative values to indicate refunds.

Comment (yuhao900914) [reply src/main/java/com/amplitude/Event.java:250]: Oh, I just realized we send the payload to api v2 eventually. Will update this to if have revenue or have price and valid quantity will pass other revenue related values.

File: src/main/java/com/amplitude/Event.java (hunk 2 above)
Comment (yuhao900914) [reply src/main/java/com/amplitude/Event.java:244]: Sure, will add the test.

File: src/main/java/com/amplitude/Event.java (hunk 1 above)
Comment (qingzhuozhen) [reply src/main/java/com/amplitude/Event.java:250]: `if have revenue or have price and valid quantity` this is already true when we get to this part right? What's our normal way to calculate this in SDK?

**Review by qingzhuozhen — APPROVED** [review]

**Pull request merged**